frontend/src/components/SignUp/SignUp.js

```diff
@@ -15,78 +21,95 @@ class SignUp extends Component {
     }));
   }
 
+  handleChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  handleSubmit = (event, changeCreatedState) => {
+    event.preventDefault();
+    createUser(this.state)
+      .then((response) => {
+        changeCreatedState(response.status);
+      })
+      .catch((error) => {
+        if (error.message.includes('422')) {
+          const parentForm = document.getElementsByTagName('form')[0];
+          const errorDiv = document.createElement('p');
+          errorDiv.setAttribute('class', 'errorMessage sign-up');
+          errorDiv.innerHTML = 'A user with that email already exists! Try signing in.';
+          parentForm.insertAdjacentElement('beforeend', errorDiv);
```

Comment:
I think instead of using vanilla JS here, we could just have a div with all of these attributes already on the page, and store the error message in state/render it from there? Or even use a `renderError` function that checks to see if `this.state.errorMessage` is empty, and if not returns a div with the error message. If you think that's doable on your time table pre-demo, that'd be awesome. Otherwise, we can just do this and come back to it.

Reply:
That's what I wanted to do . . . I'll add it to the list of things I have to continue/refactor the work I did.